### ISSUE TITLE
Runtime error on Arch Linux

### ISSUE THREAD

**Issue by mitchmindtree** [RustAudio/rust-portaudio#36]
FreeFull tried out the portaudio example today and was getting an overflow error. Here's his full output:

```
freefull@freefull-hp ~/c/o/r/rust-portaudio> cargo run --example main
     Running `target/examples/main`
PortAudio version : 1899
PortAudio version text : PortAudio V19-devel (built Jan 31 2014 21:19:35)
ALSA lib pcm.c:2239:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2239:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2239:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_route.c:947:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:947:(find_matching_chmap) Found no matching channel map
ALSA lib pulse.c:243:(pulse_connect) PulseAudio: Unable to connect: Connection refused

ALSA lib pulse.c:243:(pulse_connect) PulseAudio: Unable to connect: Connection refused

ALSA lib pulse.c:243:(pulse_connect) PulseAudio: Unable to connect: Connection refused

ALSA lib pulse.c:243:(pulse_connect) PulseAudio: Unable to connect: Connection refused

connect(2) call to /dev/shm/jack-1000/default/jack_0 failed (err=No such file or directory)
attempt to connect to server failed
Successfully initialized PortAudio
PortAudio host count : 2
PortAudio default host : 0
PortAudio host name : ALSA
PortAudio type id : -9979
Default input device info :
        version : 2
        name : default
        max input channels : 128
        max output channels : 128
        default sample rate : 48000
Default output device name : default
Successfully opened the stream.
Successfully started the stream.
Read stream available with 940 frames.
Write stream available with 1881 frames.
Read stream available with 684 frames.
Write stream available with 1625 frames.
Read stream available with 428 frames.
Write stream available with 1369 frames.
Read stream available with 172 frames.
Write stream available with 1113 frames.
Read stream available with 856 frames.
Write stream available with 876 frames.
Read stream available with 600 frames.
Write stream available with 634 frames.
Read stream available with 344 frames.
Write stream available with 393 frames.
Read stream available with 88 frames.
Write stream available with 1957 frames.
Read stream available with 1712 frames.
Write stream available with 1701 frames.
Read stream available with 1456 frames.
Write stream available with 1445 frames.
Read stream available with 1200 frames.
Write stream available with 1189 frames.
Read stream available with 944 frames.
Write stream available with 933 frames.
Read stream available with 1628 frames.
Write stream available with 677 frames.
Read stream available with 1372 frames.
Write stream available with 421 frames.
Read stream available with 1116 frames.
Write stream available with 165 frames.
Read stream available with 860 frames.
Write stream available with 1790 frames.
An error occurred while writing to the output stream: The output stream has overflowed
Successfully closed the stream.

Successfully terminated PortAudio.
```

Perhaps there is still some work we need to do to get things running smoothly on Linux?

**Comment by mitchmindtree** [comment]
This *should* be fixed by #77, #78 and #81. Will reopen this if anyone else reports further issues.

**Comment by FreeFull** [comment]
Unfortunately, now behaviour is even worse. The blocking example terminates with `fish: Job 1, “./blocking ” terminated by signal SIGSEGV (Address boundary error)` and the non-blocking example floods the console with underrun messages.
